src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/category/CategoriesFieldset.java

```
@@ -53,39 +53,71 @@ public boolean canEdit() {
 
   @CheckForNull
   public CategoryPill findCategoryWith(final String categoryName) {
-    CategoryPill result = null;
+    WebElement category = getCategoryWith(categoryName);
+    return category == null ? null : new CategoryPill(category);
+  }
 
-    for (int i = 0; i < categories.size() && result == null; i++) {
-      WebElement category = categories.get(i);
-      if (category.getText().equals(categoryName)) {
-        result = new CategoryPill(category, i);
+  /**
+   *
+   * @param categoryName to match
+   * @return first WebElement category that matches the name
+   */
+  private WebElement getCategory(List<WebElement> categoryList, final String categoryName) {
+    List<WebElement> foundCategories = categoryList.stream()
+      .filter(element -> element
+        .getAttribute("innerText")
+        .trim()
+        .equalsIgnoreCase(categoryName)
+      )
+      .collect(Collectors.toList());
+
+      // because of differences in mobile and desktop views
+      if (foundCategories.isEmpty()) {
+        return null;
```

Perhaps we can throw an exception here?